### PR TITLE
WrapperPlayServerResourcePackSend

### PR DESCRIPTION
**Pull request by amnoah**
I think I did it correctly, please let me know if anything is wrong.

Basically copied from MCProtocolLib

### CODE REVIEW

File: api/src/main/java/com/github/retrooper/packetevents/wrapper/play/server/WrapperPlayServerResourcePackSend.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        prompt = wrapper.prompt;
+    }
+
+    @Override
+    public void write() {
+        writeString(url);
+        writeString(hash);
+        writeBoolean(required);
+        writeBoolean(prompt != null);
+        writeComponent(prompt);
```
Comment (MWHunter) [inline api/src/main/java/com/github/retrooper/packetevents/wrapper/play/server/WrapperPlayServerResourcePackSend.java:70]: prompt isn't written if it is null

File: api/src/main/java/com/github/retrooper/packetevents/wrapper/play/server/WrapperPlayServerResourcePackSend.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        this.hash = hash;
+        this.required = required;
+        this.prompt = prompt;
+    }
+
+    @Override
+    public void read() {
+        url = readString();
+        hash = readString();
+        required = readBoolean();
```
Comment (MWHunter) [inline api/src/main/java/com/github/retrooper/packetevents/wrapper/play/server/WrapperPlayServerResourcePackSend.java:48]: https://wiki.vg/index.php?title=Protocol&oldid=16681#Resource_Pack_Send

1.16 and below doesn't have the required or prompt

**Review by MWHunter — CHANGES_REQUESTED** [review]

File: api/src/main/java/com/github/retrooper/packetevents/wrapper/play/server/WrapperPlayServerResourcePackSend.java (hunk 2 above)
Comment (amnoah) [reply api/src/main/java/com/github/retrooper/packetevents/wrapper/play/server/WrapperPlayServerResourcePackSend.java]: Fixed in latest commit, Required/hasPrompt/Prompt will no longer be accessed unless the server is above or equal to version 1.16.

File: api/src/main/java/com/github/retrooper/packetevents/wrapper/play/server/WrapperPlayServerResourcePackSend.java (hunk 1 above)
Comment (amnoah) [reply api/src/main/java/com/github/retrooper/packetevents/wrapper/play/server/WrapperPlayServerResourcePackSend.java]: Fixed this by adding "hasPrompt" boolean (following Wiki.vg names)

**Comment by MWHunter** [comment]
Could be better with another constructor for pre-1.17 servers but I really don't care and that can be added later when we figure out how to make better constructors that don't require every argument

**Pull request merged**